aws-xray-recorder-sdk-core/src/test/java/com/amazonaws/xray/emitters/UDPEmitterTest.java

```diff
@@ -1,21 +1,44 @@
 package com.amazonaws.xray.emitters;
 
 import com.amazonaws.xray.config.DaemonConfiguration;
-import org.junit.Assert;
+import com.amazonaws.xray.entities.DummySegment;
 import org.junit.Test;
 
 import java.net.SocketException;
 
+import static com.amazonaws.xray.AWSXRay.getGlobalRecorder;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
 public class UDPEmitterTest {
 
     @Test
     public void testCustomAddress() throws SocketException {
         String address = "123.4.5.6:1234";
-        DaemonConfiguration config = new DaemonConfiguration();
-        config.setDaemonAddress(address);
+        DaemonConfiguration config = getDaemonConfiguration(address);
 
         UDPEmitter emitter = new UDPEmitter(config);
 
-        Assert.assertEquals(address, emitter.getUDPAddress());
+        assertEquals(address, emitter.getUDPAddress());
+    }
+
+
+    @Test
+    public void sendingSegmentShouldNotThrowExceptions() throws SocketException {
+        DaemonConfiguration config = getDaemonConfiguration("host:1234");
+        UDPEmitter emitter = new UDPEmitter(config);
+
+        try {
+            final boolean success = emitter.sendSegment(new DummySegment(getGlobalRecorder()));
```

Comment:
Could you mock the UDPEmitter's datagram socket so that this unit test does not actually use the network? Adding a new package-private constructor or datagram socket setter is fine.

Reply:
Unfortunately from the stacktrace we can see it's the `DatagramPacket` constructor itself that throws, so it doesn't seem possible to completely avoid the network. Definitely preferable if we could, but for now since it didn't add value I went ahead and reverted the change to add a new constructor, we can bring it back if we find a different use case for it.

Reply:
Ah I see, thanks for fixing that. Didn't think about how mocking could obfuscate the source of exception.